File: member/mall-api/src/main/java/com/laf/mall/api/dto/RaffleInfo.java
package com.laf.mall.api.dto;

import com.fasterxml.jackson.annotation.JsonView;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;
import java.util.List;

@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class RaffleInfo {
    public interface RaffleInfoView {
    }

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private int raffle_id;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private String title;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private String picture;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private Date raffle_time_start;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private Date raffle_time_end;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private int required_points;

    private int intro_id;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private String intro;

    private boolean is_active;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private int mall_id;

    private int times;

//    private int daily;

    private String activity_code;

    @JsonView(RaffleInfo.RaffleInfoView.class)
    private List<RaffleReward> rewards;

    public long getRaffle_time_start() {
        return raffle_time_start == null ? 0L : raffle_time_start.getTime();
    }

    public long getRaffle_time_end() {
        return raffle_time_end == null ? 0L : raffle_time_end.getTime();
    }
}

File: member/manager-app/src/main/java/com/laf/manager/dto/RaffleInfo.kt
package com.laf.manager.dto

data class RaffleInfo(
        var raffle_id: Int,
        var title: String,
        var raffle_time_start: Long,
        var raffle_time_end: Long,
        var required_points: Int,
        var intro_id: Int,
        var intro: String,
        var mall_id: Int,
        var is_active: Boolean,
        var times: Int,
        var daily: Boolean,
        var activity_code: String,
        var picture: String?,
        var rewards: List<RaffleReward>
) {
    constructor(
            raffle_id: Int,
            title: String,
            raffle_time_start: Long,
            raffle_time_end: Long,
            required_points: Int,
            intro_id: Int,
            intro: String,
            mall_id: Int,
            times: Int,
            activity_code: String
    ) : this(raffle_id, title, raffle_time_start, raffle_time_end, required_points,intro_id,intro,
            mall_id,true,times,false,activity_code, null, emptyList())

    constructor(
        raffle_id: Int,
        title: String,
        raffle_time_start: Long,
        raffle_time_end: Long,
        required_points: Int,
        activity_code: String
    ) : this(raffle_id, title, raffle_time_start, raffle_time_end, required_points, 0, "", 0, false, 0, false, activity_code, "", emptyList())
}



